pkg/providers/tinkerbell/upgrade.go

```diff
@@ -134,7 +134,23 @@ func (p *Provider) SetupAndValidateUpgradeCluster(ctx context.Context, cluster *
 		return err
 	}
 
-	return p.validateAvailableHardwareForUpgrade(ctx, currentClusterSpec, clusterSpec)
+	if err := p.validateAvailableHardwareForUpgrade(ctx, currentClusterSpec, clusterSpec); err != nil {
+		return err
+	}
+
+	if p.clusterConfig.IsManaged() {
+		if err := p.applyHardwareUpgrade(ctx, clusterSpec.ManagementCluster); err != nil {
+			return err
+		}
+		if len(p.catalogue.AllBMCs()) > 0 {
```

Comment:
This check might not work here. During upgrade we do not add the existing BMCs back into the catalogue. This means `if len(p.catalogue.AllBMCs()) > 0` would be `true` only if a new hardware CSV with BMCs are fed during workload cluster upgrades.

We could do a check like https://github.com/aws/eks-anywhere/blob/main/pkg/providers/tinkerbell/upgrade.go#L190